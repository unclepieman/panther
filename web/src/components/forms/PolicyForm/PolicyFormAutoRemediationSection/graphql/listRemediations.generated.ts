/**
 * Panther is a Cloud-Native SIEM for the Modern Security Team.
 * Copyright (C) 2020 Panther Labs Inc
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */

import * as Types from '../../../../../../__generated__/schema';

import { GraphQLError } from 'graphql';
import gql from 'graphql-tag';
import * as ApolloReactCommon from '@apollo/client';
import * as ApolloReactHooks from '@apollo/client';

export type ListRemediationsVariables = {};

export type ListRemediations = Pick<Types.Query, 'remediations'>;

export const ListRemediationsDocument = gql`
  query ListRemediations {
    remediations
  }
`;

/**
 * __useListRemediations__
 *
 * To run a query within a React component, call `useListRemediations` and pass it any options that fit your needs.
 * When your component renders, `useListRemediations` returns an object from Apollo Client that contains loading, error, and data properties
 * you can use to render your UI.
 *
 * @param baseOptions options that will be passed into the query, supported options are listed on: https://www.apollographql.com/docs/react/api/react-hooks/#options;
 *
 * @example
 * const { data, loading, error } = useListRemediations({
 *   variables: {
 *   },
 * });
 */
export function useListRemediations(
  baseOptions?: ApolloReactHooks.QueryHookOptions<ListRemediations, ListRemediationsVariables>
) {
  return ApolloReactHooks.useQuery<ListRemediations, ListRemediationsVariables>(
    ListRemediationsDocument,
    baseOptions
  );
}
export function useListRemediationsLazyQuery(
  baseOptions?: ApolloReactHooks.LazyQueryHookOptions<ListRemediations, ListRemediationsVariables>
) {
  return ApolloReactHooks.useLazyQuery<ListRemediations, ListRemediationsVariables>(
    ListRemediationsDocument,
    baseOptions
  );
}
export type ListRemediationsHookResult = ReturnType<typeof useListRemediations>;
export type ListRemediationsLazyQueryHookResult = ReturnType<typeof useListRemediationsLazyQuery>;
export type ListRemediationsQueryResult = ApolloReactCommon.QueryResult<
  ListRemediations,
  ListRemediationsVariables
>;
export function mockListRemediations({
  data,
  variables,
  errors,
}: {
  data: ListRemediations;
  variables?: ListRemediationsVariables;
  errors?: GraphQLError[];
}) {
  return {
    request: { query: ListRemediationsDocument, variables },
    result: { data, errors },
  };
}
