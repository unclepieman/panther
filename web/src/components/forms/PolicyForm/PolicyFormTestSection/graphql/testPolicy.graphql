mutation TestPolicy($input: TestPolicyInput!) {
  testPolicy(input: $input) {
    results {
      id
      name
      passed
      error {
        message
      }
      functions {
        policyFunction {
          ...TestFunctionResult
        }
      }
    }
  }
}
