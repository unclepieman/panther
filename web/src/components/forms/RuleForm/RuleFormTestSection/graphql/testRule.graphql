mutation TestRule($input: TestRuleInput!) {
  testRule(input: $input) {
    results {
      id
      name
      passed
      error {
        message
      }
      functions {
        ruleFunction {
          ...TestFunctionResult
        }
        titleFunction {
          ...TestFunctionResult
        }
        dedupFunction {
          ...TestFunctionResult
        }
        alertContextFunction {
          ...TestFunctionResult
        }
        descriptionFunction {
          ...TestFunctionResult
        }
        destinationsFunction {
          ...TestFunctionResult
        }
        referenceFunction {
          ...TestFunctionResult
        }
        runbookFunction {
          ...TestFunctionResult
        }
        severityFunction {
          ...TestFunctionResult
        }
      }
    }
  }
}
