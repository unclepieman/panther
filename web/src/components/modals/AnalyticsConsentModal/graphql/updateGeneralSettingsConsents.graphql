mutation UpdateGeneralSettingsConsents($input: UpdateGeneralSettingsInput!) {
    updateGeneralSettings(input: $input) {
        email
        errorReportingConsent
        analyticsConsent
    }
}