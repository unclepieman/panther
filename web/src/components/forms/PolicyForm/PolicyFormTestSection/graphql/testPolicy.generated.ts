/**
 * Panther is a Cloud-Native SIEM for the Modern Security Team.
 * Copyright (C) 2020 Panther Labs Inc
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */

import * as Types from '../../../../../../__generated__/schema';

import { TestFunctionResult } from '../../../../../graphql/fragments/TestFunctionResult.generated';
import { GraphQLError } from 'graphql';
import gql from 'graphql-tag';
import * as ApolloReactCommon from '@apollo/client';
import * as ApolloReactHooks from '@apollo/client';

export type TestPolicyVariables = {
  input: Types.TestPolicyInput;
};

export type TestPolicy = {
  testPolicy: {
    results: Array<
      Pick<Types.TestPolicyRecord, 'id' | 'name' | 'passed'> & {
        error?: Types.Maybe<Pick<Types.Error, 'message'>>;
        functions: { policyFunction: TestFunctionResult };
      }
    >;
  };
};

export const TestPolicyDocument = gql`
  mutation TestPolicy($input: TestPolicyInput!) {
    testPolicy(input: $input) {
      results {
        id
        name
        passed
        error {
          message
        }
        functions {
          policyFunction {
            ...TestFunctionResult
          }
        }
      }
    }
  }
  ${TestFunctionResult}
`;
export type TestPolicyMutationFn = ApolloReactCommon.MutationFunction<
  TestPolicy,
  TestPolicyVariables
>;

/**
 * __useTestPolicy__
 *
 * To run a mutation, you first call `useTestPolicy` within a React component and pass it any options that fit your needs.
 * When your component renders, `useTestPolicy` returns a tuple that includes:
 * - A mutate function that you can call at any time to execute the mutation
 * - An object with fields that represent the current status of the mutation's execution
 *
 * @param baseOptions options that will be passed into the mutation, supported options are listed on: https://www.apollographql.com/docs/react/api/react-hooks/#options-2;
 *
 * @example
 * const [testPolicy, { data, loading, error }] = useTestPolicy({
 *   variables: {
 *      input: // value for 'input'
 *   },
 * });
 */
export function useTestPolicy(
  baseOptions?: ApolloReactHooks.MutationHookOptions<TestPolicy, TestPolicyVariables>
) {
  return ApolloReactHooks.useMutation<TestPolicy, TestPolicyVariables>(
    TestPolicyDocument,
    baseOptions
  );
}
export type TestPolicyHookResult = ReturnType<typeof useTestPolicy>;
export type TestPolicyMutationResult = ApolloReactCommon.MutationResult<TestPolicy>;
export type TestPolicyMutationOptions = ApolloReactCommon.BaseMutationOptions<
  TestPolicy,
  TestPolicyVariables
>;
export function mockTestPolicy({
  data,
  variables,
  errors,
}: {
  data: TestPolicy;
  variables?: TestPolicyVariables;
  errors?: GraphQLError[];
}) {
  return {
    request: { query: TestPolicyDocument, variables },
    result: { data, errors },
  };
}
