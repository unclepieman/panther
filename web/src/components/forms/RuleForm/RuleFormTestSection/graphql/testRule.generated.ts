/**
 * Panther is a Cloud-Native SIEM for the Modern Security Team.
 * Copyright (C) 2020 Panther Labs Inc
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */

import * as Types from '../../../../../../__generated__/schema';

import { TestFunctionResult } from '../../../../../graphql/fragments/TestFunctionResult.generated';
import { GraphQLError } from 'graphql';
import gql from 'graphql-tag';
import * as ApolloReactCommon from '@apollo/client';
import * as ApolloReactHooks from '@apollo/client';

export type TestRuleVariables = {
  input: Types.TestRuleInput;
};

export type TestRule = {
  testRule: {
    results: Array<
      Pick<Types.TestRuleRecord, 'id' | 'name' | 'passed'> & {
        error?: Types.Maybe<Pick<Types.Error, 'message'>>;
        functions: {
          ruleFunction: TestFunctionResult;
          titleFunction?: Types.Maybe<TestFunctionResult>;
          dedupFunction?: Types.Maybe<TestFunctionResult>;
          alertContextFunction?: Types.Maybe<TestFunctionResult>;
          descriptionFunction?: Types.Maybe<TestFunctionResult>;
          destinationsFunction?: Types.Maybe<TestFunctionResult>;
          referenceFunction?: Types.Maybe<TestFunctionResult>;
          runbookFunction?: Types.Maybe<TestFunctionResult>;
          severityFunction?: Types.Maybe<TestFunctionResult>;
        };
      }
    >;
  };
};

export const TestRuleDocument = gql`
  mutation TestRule($input: TestRuleInput!) {
    testRule(input: $input) {
      results {
        id
        name
        passed
        error {
          message
        }
        functions {
          ruleFunction {
            ...TestFunctionResult
          }
          titleFunction {
            ...TestFunctionResult
          }
          dedupFunction {
            ...TestFunctionResult
          }
          alertContextFunction {
            ...TestFunctionResult
          }
          descriptionFunction {
            ...TestFunctionResult
          }
          destinationsFunction {
            ...TestFunctionResult
          }
          referenceFunction {
            ...TestFunctionResult
          }
          runbookFunction {
            ...TestFunctionResult
          }
          severityFunction {
            ...TestFunctionResult
          }
        }
      }
    }
  }
  ${TestFunctionResult}
`;
export type TestRuleMutationFn = ApolloReactCommon.MutationFunction<TestRule, TestRuleVariables>;

/**
 * __useTestRule__
 *
 * To run a mutation, you first call `useTestRule` within a React component and pass it any options that fit your needs.
 * When your component renders, `useTestRule` returns a tuple that includes:
 * - A mutate function that you can call at any time to execute the mutation
 * - An object with fields that represent the current status of the mutation's execution
 *
 * @param baseOptions options that will be passed into the mutation, supported options are listed on: https://www.apollographql.com/docs/react/api/react-hooks/#options-2;
 *
 * @example
 * const [testRule, { data, loading, error }] = useTestRule({
 *   variables: {
 *      input: // value for 'input'
 *   },
 * });
 */
export function useTestRule(
  baseOptions?: ApolloReactHooks.MutationHookOptions<TestRule, TestRuleVariables>
) {
  return ApolloReactHooks.useMutation<TestRule, TestRuleVariables>(TestRuleDocument, baseOptions);
}
export type TestRuleHookResult = ReturnType<typeof useTestRule>;
export type TestRuleMutationResult = ApolloReactCommon.MutationResult<TestRule>;
export type TestRuleMutationOptions = ApolloReactCommon.BaseMutationOptions<
  TestRule,
  TestRuleVariables
>;
export function mockTestRule({
  data,
  variables,
  errors,
}: {
  data: TestRule;
  variables?: TestRuleVariables;
  errors?: GraphQLError[];
}) {
  return {
    request: { query: TestRuleDocument, variables },
    result: { data, errors },
  };
}
