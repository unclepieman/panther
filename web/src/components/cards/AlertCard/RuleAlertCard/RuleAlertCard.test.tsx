/**
 * Panther is a Cloud-Native SIEM for the Modern Security Team.
 * Copyright (C) 2020 Panther Labs Inc
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */

import {
  buildAlertSummary,
  buildAlertSummaryRuleInfo,
  buildDeliveryResponse,
  buildDestination,
  render,
  waitForElementToBeRemoved,
} from 'test-utils';
import React from 'react';
import {
  AlertStatusesEnum,
  AlertSummaryRuleInfo,
  DestinationTypeEnum,
  SeverityEnum,
} from 'Generated/schema';
import urls from 'Source/urls';
import { mockListDestinations } from 'Source/graphql/queries';
import RuleAlertCard from './index';

describe('RuleAlertCard', () => {
  it('should match snapshot', async () => {
    const alert = buildAlertSummary();

    const { container } = render(<RuleAlertCard alert={alert} />);

    expect(container).toMatchSnapshot();
  });

  it('displays the correct Alert data in the card', async () => {
    const alert = buildAlertSummary({
      detection: buildAlertSummaryRuleInfo({
        logTypes: ['alpha', 'beta'],
      }),
    });
    const detectionData = alert.detection as AlertSummaryRuleInfo;

    const { getByText, getByAriaLabel } = render(<RuleAlertCard alert={alert} />);

    expect(getByText(alert.title)).toBeInTheDocument();
    expect(getByAriaLabel(`Link to rule ${detectionData.ruleId}`)).toBeInTheDocument();
    expect(getByText('Events')).toBeInTheDocument();
    expect(getByAriaLabel(`Creation time for ${alert.alertId}`)).toBeInTheDocument();
    expect(getByText(SeverityEnum.Medium)).toBeInTheDocument();
    expect(getByText(AlertStatusesEnum.Triaged)).toBeInTheDocument();
    expect(getByAriaLabel('Change Alert Status')).toBeInTheDocument();

    detectionData.logTypes.forEach(logType => {
      expect(getByText(logType)).toBeInTheDocument();
    });
  });

  it('should not display link to Rule', async () => {
    const alert = buildAlertSummary();
    const detectionData = alert.detection as AlertSummaryRuleInfo;

    const { queryByText, queryByAriaLabel } = render(
      <RuleAlertCard alert={alert} hideRuleButton />
    );

    expect(queryByText(alert.title)).toBeInTheDocument();
    expect(queryByAriaLabel(`Link to rule ${detectionData.ruleId}`)).not.toBeInTheDocument();
  });

  it('should check links are valid', async () => {
    const alert = buildAlertSummary();
    const detectionData = alert.detection as AlertSummaryRuleInfo;

    const { getByAriaLabel } = render(<RuleAlertCard alert={alert} />);

    expect(getByAriaLabel('Link to Alert')).toHaveAttribute(
      'href',
      urls.logAnalysis.alerts.details(alert.alertId)
    );
    expect(getByAriaLabel(`Link to rule ${detectionData.ruleId}`)).toHaveAttribute(
      'href',
      urls.logAnalysis.rules.details(detectionData.ruleId)
    );
  });

  it('should render alert destinations logos', async () => {
    const outputId = 'destination-of-alert';
    const alert = buildAlertSummary({
      deliveryResponses: [buildDeliveryResponse({ outputId })],
    });
    const destination = buildDestination({ outputId, outputType: DestinationTypeEnum.Slack });
    const mocks = [mockListDestinations({ data: { destinations: [destination] } })];

    const { getByAriaLabel, getByAltText } = render(<RuleAlertCard alert={alert} />, {
      mocks,
    });
    const loadingInterfaceElement = getByAriaLabel('Loading...');
    expect(loadingInterfaceElement).toBeInTheDocument();
    await waitForElementToBeRemoved(loadingInterfaceElement);
    expect(getByAltText(`${destination.outputType} logo`)).toBeInTheDocument();
  });

  it('should render message that destination delivery is failing', async () => {
    const outputId = 'destination-of-alert';
    const alert = buildAlertSummary({
      deliveryResponses: [buildDeliveryResponse({ outputId, success: false })],
    });
    const destination = buildDestination({ outputId, outputType: DestinationTypeEnum.Slack });
    const mocks = [mockListDestinations({ data: { destinations: [destination] } })];

    const { getByAriaLabel, getByAltText } = render(<RuleAlertCard alert={alert} />, {
      mocks,
    });
    const loadingInterfaceElement = getByAriaLabel('Loading...');
    expect(loadingInterfaceElement).toBeInTheDocument();
    await waitForElementToBeRemoved(loadingInterfaceElement);
    expect(getByAltText(`${destination.outputType} logo`)).toBeInTheDocument();
    expect(getByAriaLabel('Destination delivery failure')).toBeInTheDocument();
  });
});
