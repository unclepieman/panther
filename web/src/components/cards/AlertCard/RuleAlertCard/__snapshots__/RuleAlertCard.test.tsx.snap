// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RuleAlertCard should match snapshot 1`] = `
@keyframes animation-0 {
  0% {
    -webkit-transform: rotate(0);
    -moz-transform: rotate(0);
    -ms-transform: rotate(0);
    transform: rotate(0);
  }

  100% {
    -webkit-transform: rotate(360deg);
    -moz-transform: rotate(360deg);
    -ms-transform: rotate(360deg);
    transform: rotate(360deg);
  }
}

@keyframes animation-1 {
  0% {
    stroke-dashoffset: 600;
  }

  100% {
    stroke-dashoffset: 0;
  }
}

.panther-42 {
  background-color: #1d2a3f;
  border-radius: 4px;
  overflow: hidden;
}

.panther-41 {
  background-color: transparent;
  padding: 16px;
}

.panther-40 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  height: 100%;
}

.panther-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  padding-right: 8px;
}

.panther-39 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  width: 100%;
}

.panther-8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  margin-bottom: 8px;
}

.panther-2 {
  font-weight: 500;
  margin-right: auto;
  max-width: 60%;
  word-break: break-word;
}

.panther-1 {
  -webkit-transition: color 0.1s ease-out;
  transition: color 0.1s ease-out;
  font-weight: 500;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #7fd3ff;
}

.panther-1:hover {
  color: #ace2ff;
}

.panther-1:focus {
  color: #ace2ff;
}

.panther-1:active,
.panther-1[data-active=true] {
  color: #11a5f3;
}

.panther-4 {
  font-size: 0.75rem;
  max-width: 20%;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.panther-3 {
  margin-left: 4px;
  color: #86a3c3;
}

.panther-5 {
  padding-left: 8px;
  padding-right: 8px;
  font-size: 0.75rem;
  color: #86a3c3;
}

.panther-6 {
  margin-right: 4px;
  color: #86a3c3;
}

.panther-10 {
  margin-right: auto;
}

.panther-9 {
  background-color: #141c2a;
  border-radius: 2px;
  padding-left: 6px;
  padding-right: 6px;
  padding-top: 4px;
  padding-bottom: 4px;
  font-weight: 700;
  font-size: 0.625rem;
  color: #09a084;
}

.panther-38 {
  display: grid;
  grid-gap: 8px;
  grid-template-columns: 3fr 4fr 3fr;
}

.panther-17 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-top: 8px;
}

.panther-11 {
  color: #bdbdbd;
  font-size: 0.563rem;
  font-weight: 500;
}

.panther-16 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  font-size: 0.875rem;
  font-weight: 500;
  opacity: 1;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-height: 24px;
}

.panther-15 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.panther-15>*:not(:last-child) {
  margin-right: 8px;
}

.panther-12 {
  max-width: 250px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.panther-14 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 24px;
  height: 24px;
  background-color: #4f627c;
  border-color: #4f627c;
  border-radius: 99999px;
}

.panther-14:hover {
  background-color: #11a5f3;
  border-color: #11a5f3;
}

.panther-13 {
  display: inline-block;
  vertical-align: sub;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 12px;
  height: 12px;
  color: currentColor;
}

.panther-22 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.panther-22>*:not(:last-child) {
  margin-right: 8px;
}

.panther-19 {
  width: 12px;
  height: 12px;
  background-color: hsl(221, 31%, 31%);
  border-radius: 99999px;
}

.panther-20 {
  font-size: 0.75rem;
  font-weight: 400;
  line-height: 1.5;
}

.panther-37 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
}

.panther-36 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -ms-flex-pack: end;
  -webkit-justify-content: flex-end;
  justify-content: flex-end;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
}

.panther-36>*:not(:last-child) {
  margin-right: 8px;
}

.panther-27 {
  height: 18px;
}

.panther-26 {
  display: inline-block;
  vertical-align: sub;
  -webkit-animation: animation-0 2s linear infinite;
  animation: animation-0 2s linear infinite;
  width: 18px;
  height: 18px;
}

.panther-25 {
  stroke-dashoffset: 600;
  stroke-dasharray: 300;
  stroke-width: 13;
  stroke-miterlimit: 10;
  stroke-linecap: round;
  -webkit-animation: animation-1 1.6s cubic-bezier(0.4, 0.15, 0.6, 0.85) infinite;
  animation: animation-1 1.6s cubic-bezier(0.4, 0.15, 0.6, 0.85) infinite;
  stroke: currentColor;
  fill: transparent;
}

.panther-28 {
  border: none;
  opacity: 0.6;
  border-left: 0.0625rem solid;
  height: auto;
  margin-left: 0;
  margin-right: 0;
  border-color: inherit;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
}

.panther-29 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  cursor: default;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  min-width: 85px;
  text-align: center;
  font-weight: 700;
  border-radius: 40px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  font-size: 0.75rem;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 16px;
  padding-right: 16px;
  border: 1px solid;
  border-color: #ffa500;
  background-color: transparent;
}

.panther-35 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.panther-35>*:not(:last-child) {
  margin-right: 4px;
}

.panther-31 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 85px;
}

.panther-30 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  cursor: default;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  min-width: 85px;
  text-align: center;
  font-weight: 700;
  border-radius: 40px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  font-size: 0.75rem;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 16px;
  padding-right: 16px;
  background-color: #6e3078;
}

.panther-34 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.panther-33 {
  outline: none;
  line-height: 1;
  border-radius: 99999px;
  height: 24px;
  width: 24px;
  -webkit-transition: border-color 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,background-color 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: border-color 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,background-color 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  border: 1px solid;
  border-color: #3e516b;
  background-color: transparent;
  cursor: pointer;
  color: #f6f6f6;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.panther-33:hover {
  background-color: #465973;
  border-color: #465973;
}

.panther-33:focus {
  background-color: #465973;
  border-color: #93a6c0;
}

.panther-33:active,
.panther-33[data-active=true] {
  background-color: #253852;
  border-color: #253852;
}

.panther-32 {
  display: inline-block;
  vertical-align: sub;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 16px;
  height: 16px;
  color: currentColor;
}

<div>
  <header
    id="main-header"
  />
  <section
    class="panther-42"
  >
    <div
      class="panther-41"
    >
      <div
        class="panther-40"
      >
        <div
          class="panther-0"
        />
        <div
          class="panther-39"
        >
          <header
            class="panther-8"
          >
            <h4
              class="panther-2"
            >
              <a
                aria-label="Link to Alert"
                class="panther-1"
                href="/log-analysis/alerts/f67b8f04-5fac-404a-93a4-38db29f258ba/"
              >
                indexing
              </a>
            </h4>
            <div
              class="panther-4"
            >
              240
              <span
                class="panther-3"
              >
                Events
              </span>
            </div>
            <span
              class="panther-5"
            >
              •
            </span>
            <div
              aria-label="Creation time for f67b8f04-5fac-404a-93a4-38db29f258ba"
              class="panther-4"
            >
              <span
                class="panther-6"
              >
                Created
              </span>
              2020-08-08 12:15 GMT
            </div>
          </header>
          <div
            class="panther-10"
          >
            <div
              class="panther-9"
            >
              RULE ERROR
            </div>
          </div>
          <div
            class="panther-38"
          >
            <dl
              class="panther-17"
            >
              <dt
                aria-labelledby="ruleobject-object"
                class="panther-11"
              >
                Rule
              </dt>
              <dd
                aria-labelledby="ruleobject-object"
                class="panther-16"
              >
                <div
                  class="panther-15"
                >
                  <span
                    class="panther-12"
                  >
                    8780849b-30b8-4ce2-934b-bf033369b110
                  </span>
                  <a
                    aria-label="Link to rule 8780849b-30b8-4ce2-934b-bf033369b110"
                    href="/detections/rules/8780849b-30b8-4ce2-934b-bf033369b110/"
                  >
                    <div
                      class="panther-14"
                    >
                      <svg
                        class="panther-13"
                        role="presentation"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M12 4L10.59 5.41L16.17 11H4V13H16.17L10.59 18.59L12 20L20 12L12 4Z"
                        />
                      </svg>
                    </div>
                  </a>
                </div>
              </dd>
            </dl>
            <dl
              class="panther-17"
            >
              <dt
                aria-labelledby="log-typesobject-object"
                class="panther-11"
              >
                Log Types
              </dt>
              <dd
                aria-labelledby="log-typesobject-object"
                class="panther-16"
              >
                <div
                  class="panther-22"
                >
                  <div
                    class="panther-15"
                  >
                    <span
                      class="panther-19"
                    />
                    <span
                      class="panther-20"
                    >
                      Personal Loan Account
                    </span>
                  </div>
                </div>
              </dd>
            </dl>
            <div
              class="panther-37"
            >
              <div
                class="panther-36"
              >
                <div
                  class="panther-27"
                >
                  <svg
                    aria-label="Loading..."
                    class="panther-26"
                    viewBox="0 0 150 150"
                  >
                    <circle
                      class="panther-25"
                      color="current"
                      cx="75"
                      cy="75"
                      r="60"
                    />
                  </svg>
                </div>
                <hr
                  aria-orientation="vertical"
                  class="panther-28"
                />
                <div
                  aria-atomic="true"
                  class="panther-29"
                  role="status"
                >
                  MEDIUM
                </div>
                <div
                  class="panther-35"
                >
                  <div
                    class="panther-31"
                  >
                    <div
                      aria-atomic="true"
                      class="panther-30"
                      role="status"
                    >
                      TRIAGED
                    </div>
                  </div>
                  <div
                    aria-controls="menu--1"
                    aria-haspopup="true"
                    class="panther-34"
                    data-reach-menu-button=""
                    id="menu-button--menu--1"
                    type="button"
                  >
                    <button
                      aria-label="Change Alert Status"
                      aria-pressed="false"
                      class="panther-33"
                      type="button"
                    >
                      <svg
                        class="panther-32"
                        role="presentation"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M7 10L12 15L17 10H7Z"
                        />
                      </svg>
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
  <footer
    id="footer"
  />
</div>
`;
